design-proposals/dedicated-migration-network.md

```diff
@@ -0,0 +1,94 @@
+# Overview
+This design provides a way to migrate VMIs (Virtual Machine Instances) over a dedicated network.
+VMI migrations currently happen over the main Kubernetes network all pods are connected to.
+
+## Motivation
+In order for VMI migrations to succeed, there must be a fair amount of bandwidth available.
+If for example the memory of a migrating VMI changes faster than the current speed of the main network, the migration will never succeed.
+This is especially problematic during mass eviction events like platform upgrades.
+This problem is currently mitigated by throttling the number of concurrent migrations and using long timeouts.
+
+## Goals
+- Allow a cluster admin to setup and dedicate a secondary physical cluster network to migrations.
+- Keep the main network as the default medium for migrations.
+
+## Non Goals
+- Network configuration. A functional secondary physical network should be provided.
+- Network definition. A valid NetworkAttachmentDefinition should be written by the cluster admin and provided to the KubeVirt CR (Custom Resource).
+- Cluster components. A multi-network CNI like multus should be properly installed, as well as necessary plugins, and any required daemon (such as CNI DHCP) should be running on each node.
+
+## User story
+As an administrator of the cluster I want all my VMIs to be migrated over a dedicated network to provide a stable connection with a larger bandwidth comparing to the default k8s pod network
+
+## Definition of Users
+This feature is intended for cluster admins who wish to accelerate and/or control migration network usage.
+
+## Repos
+- kubevirt/kubevirt:
+  - add a CR option and handle it at the virt-handler level
+- kubevirt/kubevirtci:
+  - install whereabouts on CNAO lanes
+
+# Design
+If a dedicated migration network is defined in the KubeVirt CR, migration target virt-handler should communicate their IP of that network instead of the main one.
+The source virt-handler will then use that IP to communicate with the target, which will get routed through the secondary network.
```

Comment:
I have some questions which will be nice to see posted in this section:
- How (from a high level) is the IP address supposed to be resolved?
- How will virt-handler know which interface is the dedicated one of the migration network? Based on the ID multus will assign it? Is this safe for future additional networks added?
- The network will use a single flat subnet, right?
- What will cause the traffic to be routed through the secondary interface? The directly connected subnet? A special route?
- Are MAC addresses managed in this solution (compared to VMI interfaces, where we do have an option to manage it)?